code-challenges.test.js

```diff
@@ -65,5 +115,37 @@ const vowelTester2 = "academy"
 const vowelTester3 = "challenges"
 // Expected output: 2
 
+describe("theVowel", () => {
+    it("takes in a string as an argument and returns the index of the first vowel of the string", () => {
+        expect(theVowel(vowelTester1)).toEqual(1)
+        expect(theVowel(vowelTester2)).toEqual(0)
+        expect(theVowel(vowelTester3)).toEqual(2)
+    })
+})
 
 // b) Create the function that makes the test pass.
+
+// Pseudo Code:
+// This program will allow the user to enter a string and return the index of the first vowel of the string.
+// Create a function expression. (theVowel)
+// Create a variable that represents vowels. (v)
+// Create a for statement within the function, that will begin at the 0 index, end at the index length of a given string, and will count up by 1 increment.
+    // Within for, create a for statement that will begin at the 0 index, end at the index length of v, and will count up by 1 increment. 
+        // Within for, create an if statement that will evaluate if the given index value of a string is equivalent to an index value of the variable. 
+        // Have the function return the index of the given value if the evaluation is true. 
+// Console log the new function with the challenge parameters.
+
+let theVowel = (string) => {
+    let v = ["a","e","i","o","u"]
+    for(i = 0; i < string.length; i++) {
+        for(j = 0; j < v.length; j++) {
+            if(string[i] == v[j]) {
+            return i
+            }
+        }
+    }
+}
```

Comment:
Ooo, nested for loops!  Many people get very confused with nested for loops, but this looks great!